nf_core/download.py

```diff
@@ -742,16 +742,24 @@ def singularity_pull_image(self, container, out_path, cache_path, progress):
         task = progress.add_task(container, start=False, total=False, progress_type="singularity_pull", current_log="")
 
         # Run the singularity pull command
-        proc = subprocess.Popen(
+        with subprocess.Popen(
             singularity_command,
             stdout=subprocess.PIPE,
             stderr=subprocess.STDOUT,
             universal_newlines=True,
             bufsize=1,
-        )
-        for line in proc.stdout:
-            log.debug(line.strip())
-            progress.update(task, current_log=line.strip())
+        ) as proc:
+            lines = []
+            for line in proc.stdout:
+                lines.append(line)
+                progress.update(task, current_log=line.strip())
+
+        if lines:
+            # something went wrong with the container retrieval
+            if any("FATAL: " in line for line in lines):
+                log.info("Singularity container retrieval fialed with the following error:")
+                log.info("".join(lines))
```

Comment:
log.info will show up in the debug log. no? so why repeat the message?

Reply:
Yes, true. I sometimes add both just to have slightly different messages, but fine for me if log.info already shows the complete error message. I think this is the only comment left, so feel free to merge it if you like :)

Reply:
Small addendum about my thought process: logging the complete external error message makes sense because we do not control it and it might become more useful with time. That's why there is only one "log level".